File: Codeforces/contest/1111/B.py
[n, k, m] = list(map(int, input().split()))
a = list(map(int, input().split()))
a.sort()
sum = 0
for i in a:
    sum += i
ans = (sum + min(k * n, m)) / n
for i in range(min(len(a) -1 , m)):
    sum -= a[i]
    tmp = sum + min(k * (n - i - 1), m - i - 1)
    ans = max(ans, tmp / (n - i - 1))
print(ans)

File: Codeforces/contest/1111/B.cpp

#include <bits/stdc++.h>
using namespace std;
const int N = 1e5 + 10;
long long  a[N];
int main() {
    //FILE *file = freopen("../input", "r", stdin);
    long long  n, k, m;
    cin >> n >> k >> m;
    long long  sum =0 ;
    for (int i = 1; i <= n; i++) {
        cin >> a[i];
        sum += a[i];

    }
	sort(a +1, a + n +1);
	long double ans = (long double)(sum + min(k*n,m))/(long double)n;
	for(int i=1;i<=min(n-1,m);i++){
		sum -= a[i];
		long double tmp = (long double)(sum + min(k*(n-i),m-i))/(long double)(n-i);
		ans = max(ans, tmp);
	}
	cout<<fixed<<setprecision(20)<<ans<<endl;
	return 0;
}
